regress/unittests/test_helper/test_helper.c

```diff
@@ -392,7 +392,7 @@ tohex(const void *_s, size_t l)
 
 	assert(r != NULL);
 	for (i = j = 0; i < l; i++) {
-		r[j++] = hex[(s[i] >> 4) & 0xf];
+		r[j++] = hex[s[i] >> 4];
```

Comment:
what warning is this fixing?

it's also probably wrong on platforms that don't have an integral 8-bit type (Cray is the only example I know of and those are pretty rare these days, but it's allowed by the standards).